src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -514,12 +515,29 @@ protected function initializeContainer()
         $cache = new ConfigCache($this->getCacheDir().'/'.$class.'.php', $this->debug);
         $fresh = true;
         if (!$cache->isFresh()) {
-            $container = $this->buildContainer();
+            if ($this->debug) {
+                $collectedLogs = array();
+                $previousHandler = set_error_handler(function () use (&$collectedLogs, &$previousHandler) {
+                    $collectedLogs[] = $args = func_get_args();
+
+                    return $previousHandler;
```

Comment:
That's not wrong, that's correct in fact: the first request logs more things than the next one.
Not calling the previous handler prevents it to throw an exception for regular non silenced warnings/notices. This is absolutely required.